### PR TITLE
CI: Improve bootstraping before each test

### PR DESCRIPTION
**Pull request by tgraf**
See individual commits

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-gke

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -1676,19 +1676,22 @@ func (kub *Kubectl) CiliumInstall(filename string, options map[string]string) er
 		return err
 	}
 
-	// Remove cilium DS to ensure that new instances of cilium-agent are started
-	// for the newly generated ConfigMap. Otherwise, the CM changes will stay
-	// inactive until each cilium-agent has been restarted.
-	if err := kub.DeleteCiliumDS(); err != nil {
```
Comment (tklauser) [inline test/helpers/kubectl.go:1690]: Looks like this was the only place where `DeleteCiliumDS` is called. Should its implementation be removed as well?

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -1676,19 +1676,22 @@ func (kub *Kubectl) CiliumInstall(filename string, options map[string]string) er
...
+		resourcesToDelete = map[string]string{
+			"cilium":          "daemonset",
+			"cilium-operator": "deployment",
+		}
+	)
 
-	// Remove cilium operator to ensure that new instances of cilium-operator are started
-	// for the newly generated ConfigMap. Otherwise, the CM changes will stay
-	// inactive until each cilium-operator has been restarted.
-	if err := kub.DeleteCiliumOperator(); err != nil {
```
Comment (tklauser) [inline test/helpers/kubectl.go:1697]: Same here. Looks like this was the only place where `DeleteCiliumOperator` is called. Remove its implementation?

**Review by tklauser — CHANGES_REQUESTED** [review]
Nice improvements! Two small nits about functions that are no longer used after this PR and could be removed.

File: test/helpers/kubectl.go [hunk 3]
```diff
@@ -1676,19 +1725,22 @@ func (kub *Kubectl) CiliumInstall(filename string, options map[string]string) er
...
-		return err
+	wg.Add(len(resourcesToDelete))
+	for resource, resourceType := range resourcesToDelete {
+		go func(resource, resourceType string) {
+			_ = kub.DeleteResource(resourceType, "-n "+GetCiliumNamespace(GetCurrentIntegration())+" "+resource)
```
Comment (nebril) [inline test/helpers/kubectl.go]: I think you can use `CiliumNamespace` instead of `GetCiliumNamespace` here, just to be consistent.

**Review by nebril — CHANGES_REQUESTED** [review]
One comment inline, agreed with tklauser comments. Otherwise looks good!

File: test/helpers/kubectl.go (hunk 1 above)
Comment (tgraf) [reply test/helpers/kubectl.go:1690]: I'm still seeing many uses:
```
test/helpers/kubectl.go:func (kub *Kubectl) DeleteCiliumDS() error {
test/k8sT/Chaos.go:             kubectl.DeleteCiliumDS()
test/k8sT/Conformance.go:               kubectl.DeleteCiliumDS()
test/k8sT/Health.go:            kubectl.DeleteCiliumDS()
test/k8sT/KafkaPolicies.go:             kubectl.DeleteCiliumDS()
test/k8sT/Nightly.go:           kubectl.DeleteCiliumDS()
test/k8sT/Policies.go:          kubectl.DeleteCiliumDS()
test/k8sT/PoliciesNightly.go:           kubectl.DeleteCiliumDS()
test/k8sT/Services.go:          kubectl.DeleteCiliumDS()
test/k8sT/Services.go:                                  kubectl.DeleteCiliumDS()
test/k8sT/demos.go:             kubectl.DeleteCiliumDS()
test/k8sT/external_ips.go:              kubectl.DeleteCiliumDS()
test/k8sT/fqdn.go:              kubectl.DeleteCiliumDS()
test/k8sT/istio.go:             kubectl.DeleteCiliumDS()
```

Are you talking about a different function?

**Comment by tgraf** [comment]
test-me-please

File: test/helpers/kubectl.go (hunk 1 above)
Comment (tklauser) [reply test/helpers/kubectl.go:1690]: Grepped for the wrong function, sorry.

**Review by tklauser — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Pull request merged**